backend/src/apiserver/resource/resource_manager.go

```diff
@@ -1274,7 +1274,45 @@ func (r *ResourceManager) GetPipelineVersionTemplate(versionId string) ([]byte,
 	return template, nil
 }
 
-func (r *ResourceManager) AuthenticateRequest(ctx context.Context) (string, error) {
+// Authorize verifies whether the user identity, which is contained in the context object,
+// can perform some action (verb) on a resource (resourceType/resourceName) living in the
+// target namespace. If the returned error is nil, the authorization passes. Otherwise,
+// authorization fails with a non-nil error.
+func (r *ResourceManager) IsAuthorized(ctx context.Context, resourceAttributes *authorizationv1.ResourceAttributes) error {
```

Comment:
@annajung After I think twice I got that `IsAuthorized` func better fit in `ResourceManager`. The idea is that  `IsAuthorized` to be public func while [authenticateRequests](https://github.com/difince/pipelines/blob/9482f5053275d560df9efd8063c0f01a638ed4a0/backend/src/apiserver/resource/resource_manager.go#L1315) and [IsRequestAuthorized](https://github.com/difince/pipelines/blob/9482f5053275d560df9efd8063c0f01a638ed4a0/backend/src/apiserver/resource/resource_manager.go#L1333) to be kept private. 

Note:  This [issue](https://github.com/kubeflow/pipelines/pull/7351/files) prohibits transforming `IsRequestAuthorized` into a private func as IsRequestAuthorized is directly used when Uploading Pipeline, but it shouldn't

Reply:
Another even better place for these three funcs we could consider in the future is the `/auth` package...